app/javascript/src/enrollment.js

```diff
@@ -1,25 +1,33 @@
-var initiallyHiddenEnrollmentPanels = document.getElementsByClassName("initially_hidden_enrollment");
-var enrollmentToggleCheckbox = document.getElementById("display_all_enrollments");
+// For when families home page is refreshed when user on it
+document.addEventListener("DOMContentLoaded", function () {
+  const initiallyHiddenEnrollmentPanels = document.getElementsByClassName("initially_hidden_enrollment");
+  const enrollmentToggleCheckbox = document.getElementById("display_all_enrollments");
+  const enrollmentToggleButton = document.getElementById("display_all_enrollments_btn");
 
-function toggleDisplayEnrollments(event) {
-    if (event.target.checked) {
-        for (var i = 0; i < initiallyHiddenEnrollmentPanels.length; i++) {
-            initiallyHiddenEnrollmentPanels[i].classList.remove("hidden");
-        }
-    } else {
-        for (var i = 0; i < initiallyHiddenEnrollmentPanels.length; i++) {
-            initiallyHiddenEnrollmentPanels[i].classList.add("hidden");
-        }
+  enrollmentToggleButton.addEventListener('click', toggleDisplayEnrollments);
+  enrollmentToggleCheckbox.addEventListener('click', toggleDisplayEnrollments);
+
+  function toggleDisplayEnrollments(event) {
+    if (event.target.type == "submit") {
+      enrollmentToggleCheckbox.checked = !enrollmentToggleCheckbox.checked;
     }
-};
 
-// For when family home page loaded through clicking off of the families index page
-if (enrollmentToggleCheckbox != null || enrollmentToggleCheckbox != undefined) {
-    enrollmentToggleCheckbox.addEventListener('click', toggleDisplayEnrollments);
-};
+    if (enrollmentToggleCheckbox.checked) {
```

Comment:
Probably more readable and concise to:
1. move the innerText write outside of the loop, and maybe even use a ternary to set it
2. maintain one loop and move the if check into the loop
Moreover, maybe a nitpick, why not use syntactic sugar for the loop?
`
for (initiallyHiddenEnrollmentPanel in initiallyHiddenEnrollmentPanels) {
  if (enrollmentToggleCheckbox.checked) {
    initiallyHiddenEnrollmentPanel.show();
  } else {
    initiallyHiddenEnrollmentPanel.hide();
  }
`